src/test/java/org/aszjch/demoapp/domain/article/port/PostgresArticleRepositoryTest.java

```diff
@@ -0,0 +1,141 @@
+package org.aszjch.demoapp.domain.article.port;
+
+import org.aszjch.demoapp.TestClockBean;
+import org.aszjch.demoapp.TestcontainersConfiguration;
+import org.aszjch.demoapp.domain.article.Article;
+import org.aszjch.demoapp.infrastructure.repository.TruncatingRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Import;
+
+import java.time.Clock;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SpringBootTest
+@Import({TestcontainersConfiguration.class, TestClockBean.class})
+class PostgresArticleRepositoryTest {
+
+    @Autowired
+    private ArticleRepository postgresArticleRepository;
+    @Autowired
+    private TruncatingRepository truncatingRepository;
+    @Autowired
+    private Clock clock;
+
+    @BeforeEach
+    void cleanTable() {
+        truncatingRepository.truncateTable();
+    }
+
+    @Test
+    void saves_article() {
+        Article article = new Article();
+        article.setTitle("Test title");
+        article.setAuthor("Test author");
+        article.setAbstractText("Test abstract text");
+        article.setCreationDate(OffsetDateTime.now(clock));
+
+        Article saved = postgresArticleRepository.save(article);
+
+        assertNotNull(saved);
+        assertAll(
+                () -> assertNotNull(saved.getId()),
+                () -> assertEquals(article.getTitle(), saved.getTitle()),
+                () -> assertEquals(article.getAuthor(), saved.getAuthor()),
+                () -> assertEquals(article.getAbstractText(), saved.getAbstractText()),
+                () -> assertEquals(OffsetDateTime.now(clock), saved.getCreationDate()),
+                () -> assertNull(saved.getFilename()));
+    }
+
+    @Test
+    void finds_article_by_id() {
+        Article article = new Article();
```

Comment:
Ten artykuł powinien być wczytany już na BD. Nie powinno się w testach korzystać z metod które też testujemy. Powinniśmy testcontainerki wykorzystać do wgrania odpowiednich data setów testowych.
To co tutaj robimy jest antypatternem, lepiej się tego pozbyć.